[docs] Label all demos and document labeling (#3635)
mui/base-ui: docs/src/app/(docs)/react/components/checkbox/page.mdx

## Examples

### Labeling a checkbox

Label checkboxes using an enclosing `<label>` element that wraps `<Checkbox.Root>`. Enclosing labels ensure that clicking on any gaps between the label and checkbox still toggles the control, and announce the label correctly to screen readers.

> The [Field](/react/components/field) component can also be used to provide accessible labels for individual checkbox without needing to keep track of manual `id` or `aria-label` associations. It supports both enclosing and sibling labeling patterns, along with a UX improvement to prevent double clicks from selecting the label text.

Reviewer: Grammarly suggestion.
```suggestion
The [Field](/react/components/field) component can also be used to provide accessible labels for individual checkboxes, eliminating the need to track `id` or `aria-label` associations manually. It supports both enclosing and sibling labeling patterns, along with a UX improvement to prevent double clicks from selecting the label text.
```
